tests-build/tests/fail/macros_invalid_input.stderr

```diff
@@ -1,77 +1,59 @@
   |
 9 | #[tokio::main(threadpool::bar)]
-  |               ^^^^^^^^^^^^^^^
+  |               ^^^^^^^^^^
 
-error: the `async` keyword is missing from the function declaration
-  --> $DIR/macros_invalid_input.rs:13:1
+error: functions marked with `#[tokio::test]` must be `async`
+  --> tests/fail/macros_invalid_input.rs:13:1
    |
 13 | fn test_is_not_async() {}
    | ^^
 
-error: Unknown attribute foo is specified; expected one of: `flavor`, `worker_threads`, `start_paused`
-  --> $DIR/macros_invalid_input.rs:15:15
+error: unknown option `foo`, expected one of: `flavor`, `worker_threads`, `start_paused`
+  --> tests/fail/macros_invalid_input.rs:15:15
    |
 15 | #[tokio::test(foo)]
    |               ^^^
 
-error: Unknown attribute foo is specified; expected one of: `flavor`, `worker_threads`, `start_paused`
-  --> $DIR/macros_invalid_input.rs:18:15
+error: unknown option `foo`, expected one of: `flavor`, `worker_threads`, `start_paused`
+  --> tests/fail/macros_invalid_input.rs:18:15
    |
 18 | #[tokio::test(foo = 123)]
-   |               ^^^^^^^^^
+   |               ^^^
 
-error: Failed to parse value of `flavor` as string.
-  --> $DIR/macros_invalid_input.rs:21:24
+error: no such runtime flavor, the runtime flavors are: "current_thread", "multi_thread"
+  --> tests/fail/macros_invalid_input.rs:21:24
    |
 21 | #[tokio::test(flavor = 123)]
    |                        ^^^
 
-error: No such runtime flavor `foo`. The runtime flavors are `current_thread` and `multi_thread`.
-  --> $DIR/macros_invalid_input.rs:24:24
+error: no such runtime flavor, the runtime flavors are: "current_thread", "multi_thread"
+  --> tests/fail/macros_invalid_input.rs:24:24
    |
 24 | #[tokio::test(flavor = "foo")]
    |                        ^^^^^
 
-error: The `start_paused` option requires the `current_thread` runtime flavor. Use `#[tokio::test(flavor = "current_thread")]`
-  --> $DIR/macros_invalid_input.rs:27:55
-   |
-27 | #[tokio::test(flavor = "multi_thread", start_paused = false)]
-   |                                                       ^^^^^
-
-error: Failed to parse value of `worker_threads` as integer.
-  --> $DIR/macros_invalid_input.rs:30:57
-   |
-30 | #[tokio::test(flavor = "multi_thread", worker_threads = "foo")]
-   |                                                         ^^^^^
-
-error: The `worker_threads` option requires the `multi_thread` runtime flavor. Use `#[tokio::test(flavor = "multi_thread")]`
-  --> $DIR/macros_invalid_input.rs:33:59
+error: the `worker_threads` option requires the "multi_thread" runtime flavor. Use `#[tokio::test(flavor = "multi_thread")]`
+  --> tests/fail/macros_invalid_input.rs:33:59
    |
 33 | #[tokio::test(flavor = "current_thread", worker_threads = 4)]
    |                                                           ^
 
-error: second test attribute is supplied
-  --> $DIR/macros_invalid_input.rs:37:1
-   |
-37 | #[test]
-   | ^^^^^^^
-
 error: duplicated attribute
```

Comment:
I would suggest denying `duplicate_macro_attributes` lint in `macros_invalid_input.rs` to generate a consistent stderr file regardless of whether `-D warnings` is specified or not.

Reply:
Good idea!